helios-tools/src/main/java/com/spotify/helios/cli/command/HostListCommand.java

```diff
@@ -108,6 +116,12 @@ int run(final Namespace options, final HeliosClient client, final PrintStream ou
     }
 
     final List<String> sortedHosts = natural().sortedCopy(hosts);
+    final List<String> selectedLabels = options.getList(labelArg.getDest());
+
+    if (selectedLabels != null && !selectedLabels.isEmpty() && json) {
+      System.err.println("Warning: filtering by label is not supported for JSON output. Not doing"
```

Comment:
It's more because we need the host status to filter, and right now we don't get the host status if all you ask for is JSON output. We could add this functionality, but if you're doing JSON output you can always filter it yourself with `jq` or similar anyway.